### PR TITLE
make publishing introsets optional 

### PR DESCRIPTION
**Pull request by majestrate**
using `reachable=false` to disable publishing introsets

### CODE REVIEW

**Comment by majestrate** [comment]
should it be different for windows?

**Comment by notlesh** [comment]
> should it be different for windows?

Default to true? (e.g. don't publish introsets by default on windows)

**Comment by jagerman** [comment]
Maybe it should be default true everywhere.

**Comment by majestrate** [comment]
`reachable=true` is the default and having a double negative like `no-publish=false` is weird.

**Comment by jagerman** [comment]
Whoops, I meant default false everywhere (i.e. no reachable).

**Comment by majestrate** [comment]
what if we make it tied to using a persistent address? i.e. if `keyfile` is set reachable, otherwise not reachable unless explicitly specified.

**Comment by jagerman** [comment]
That sounds good.  Also I don't think we want the option to apply at all to snapps: they should always be reachable.

**Comment by majestrate** [comment]
updated

File: llarp/handlers/tun.cpp [hunk 1]
```diff
@@ -118,6 +118,17 @@ namespace llarp
...
+          m_PublishIntroSet = true;
+          LogInfo(Name(), " setting to be reachable by default");
+        }
+        else
+        {
+          LogError(Name(), " config option reachable = '", v,
+                   "' does not make sense");
+          return false;
+        }
       }
```
Comment (notlesh) [inline llarp/handlers/tun.cpp]: This is more of an observation, but we now have >= 15 lines of code to handle the 'reachable' config option... Seems like we may be missing some DRY (etc.) somewhere

File: llarp/service/context.cpp [hunk 2]
```diff
@@ -149,7 +149,12 @@ namespace llarp
...
       configOpts.push_back({"type", DefaultEndpointType()});
       // non reachable by default as this is the default endpoint
-      configOpts.push_back({"reachable", "false"});
+      // but only if no keyfile option provided
+      if(opts.count("keyfile") == 0)
+      {
+        configOpts.push_back({"reachable", "false"});
+      }
+
       {
```
Comment (notlesh) [inline llarp/service/context.cpp]: Will this effectively override an explicit `reachable=true`?

Comment (majestrate) [reply llarp/service/context.cpp]: only if it's not provided by user

Comment (majestrate) [reply llarp/service/context.cpp]: (no)

**Pull request merged**